src/test/java/teammates/test/cases/logic/FeedbackSessionsLogicTest.java

```diff
@@ -1446,9 +1446,9 @@ private FeedbackSessionAttributes getNewFeedbackSession() {
         return FeedbackSessionAttributes.builder("fsTest1", "testCourse", "valid@email.tmt")
                 .withCreatedTime(Instant.now())
                 .withStartTime(Instant.now())
-                .withEndTime(Instant.now())
                 .withSessionVisibleFromTime(Instant.now())
```

Comment:
Visible should before start?

Actually, why do we want to swap the order to make the session valid? It is totally fine to make use of methods inside `TimeHelperExtension` to make a valid session. e.g. Session visible from 3 hour ago, start from 2 hour ago, end 1 hour ago and visible now.

Reply:
> Visible should before start?

that will also work as the session will be first visible than start.

```suggestion
                .withSessionVisibleFromTime(Instant.now())
                .withStartTime(Instant.now())
```
> why do we want to swap the order to make the session valid? It is totally fine to make use of methods inside `TimeHelperExtension` to make a valid session.

i am not sure what do you want to say here , do you mean to use methods like 
`public static Instant getInstantMinutesOffsetFromNow(long offsetInMinutes)` and `public static Instant getInstantHoursOffsetFromNow(long offsetInHours)`  from `TimeHelperExtension` to return an `Instant` and pass it to 
`withStartTime` and `visible` etc methods ?

Reply:
> that will also work as the session will be first visible than start.

This is a "must" condition. A session must be visible before it can be started.

> i am not sure what do you want to say here 

If we insist to use `Instant.now()`, we need to adjust the order carefully and it might be error-prone in the future. I am saying why not using

```java
...
.withSessionStartTime(TimeHelperExtension. getInstantHoursOffsetFromNow(-1))
.withSessionEndTime(TimeHelperExtension. getInstantHoursOffsetFromNow(0))
...
```
to construct a valid session. No definitive order is required and it will always be a valid session no matter how slow/fast the code is executed.

Reply:
i have one more idea , why don't We create an `Instant startingTime`  and assign it `Instant.now()` right before returning the `feedbackSessionAttribute`  and using that `startingTime`  as an parameter for all starting methods like  
``` java
     .withCreatedTime(startingTime)
     .withSessionStartTime(startingTime)
     .withStartTime(startingTime)
```
and in other we can simply use `Instant.now()` as a parameter,
this way session would be both valid and concise.

or otherwise we can do it your way
>...
.withSessionStartTime(TimeHelperExtension. getInstantHoursOffsetFromNow(-1))
.withSessionEndTime(TimeHelperExtension. getInstantHoursOffsetFromNow(0))
...

Reply:
> this way session would be both valid and concise.

Though I don't see concise it can be.

So how about other times? (`endTime`? `resultVisibleTime`? )

Reply:
> So how about other times?

we will simply use `Instant.now()` for other times ,this way we will get time when they are actually reached in code (no matter how slow/fast code works )

Reply:
should i do that or anything else ?

Reply:
You can push your changes first.